src/path_manager.c

```diff
@@ -126,9 +126,24 @@ static char const *get_pm_name(void const *data, size_t len)
 }
 #endif // MPTCPD_ENABLE_PM_NAME
 
+/**
+ * @brief Initialize @c mptcpd_addr instance.
+ *
+ * Initialize a @c mptcpd_addr instance with the provided IPv4 or IPv6
+ * address.  Only one is required and used.
+ *
+ * The port will be set to zero if a no port is provided, i.e. @a port
+ * is @c NULL.  That may be the case for MPTCP generic netlink API
+ * events where the port is optional.
+ *
+ * @param[in]     addr4 IPv4 internet address.
+ * @param[in]     addr6 IPv6 internet address.
+ * @param[in]     port  IP port.
+ * @param[in,out] addr  mptcpd network address information.
+ */
 static void get_mptcpd_addr(struct in_addr const *addr4,
                             struct in6_addr const *addr6,
-                            in_port_t port,
+                            in_port_t const *port,
```

Comment:
I'm curious: what's the benefit of passing by reference here?

Reply:
It was either that or declare a temporary `port` variable in `handle_new_addr()`, e.g.:
```diff
+        in_port_t _port = 0;
+
+        if (port != NULL)
+                _port = *port;
+
         struct mptcpd_addr addr;
-        get_mptcpd_addr(addr4, addr6, *port, &addr);
+        get_mptcpd_addr(addr4, addr6, _port, &addr);
```
I didn't want to add a temporary, but I suppose it's not as bad as I thought it would be.  The patch is certainly smaller.  Would you prefer this approach?

Reply:
Ok - I hadn't noticed that in `handle_new_addr()`. It's up to you, it just caught my eye because I'm more used to passing a 16-bit value for an `[in]` parameter.

Could avoid the temp var: `get_mptcpd_addr(addr4, addr6, port ? *port : 0, &addr);`

Reply:
I thought you were averse to the ternary operator for some reason, so I didn't consider it.  :)

Let's go with that.

Reply:
Ha! Averse to nested ternary operators :smile: